pkg/meta/parse.go

```diff
@@ -332,9 +332,12 @@ func NewManifestData(repoName string, manifestBlob []byte, imageStore storageTyp
 		return mTypes.ManifestData{}, err
 	}
 
-	err = json.Unmarshal(configBlob, &configContent)
-	if err != nil {
-		return mTypes.ManifestData{}, err
+	// with artifacts support, config can be used to store the actual artifact so wont unmarshal
+	if manifestContent.Config.MediaType == ispec.MediaTypeImageConfig {
```

Comment:
Thinking there could be issues if the non-image-config is persisted in the db and unmarshalled later when making the search reads. I think this case should be handled when reading from the DB as well. If we're not doing it already.